### ISSUE TITLE
No access to device logs

### ISSUE THREAD

**Issue by Steve-Mcl** [flowforge/flowforge#2620]
### Current Behavior

![image](https://github.com/flowforge/flowforge/assets/44235289/e9453745-0e87-4e5d-b41d-a98938e14ff2)

![image](https://github.com/flowforge/flowforge/assets/44235289/bff9310b-ea1e-4355-b1e8-c8e866aab0b1)

![image](https://github.com/flowforge/flowforge/assets/44235289/8d9c754d-b0fb-4303-8fe6-0477f3e29777)

"Device Logs" entry is correctly added to the `navigation` array - BUT - I suspect the `SectionNavigationHeader` component doesnt support dynamic tabs. 

(The Device Logs entry is added to the navigation array after the page loads / behind a feature flag: https://github.com/flowforge/flowforge/blob/e225b7457d6683bb148f5b5969ac404e3e78b57f/frontend/src/pages/device/index.vue#L147-L152

### Expected Behavior

Access to device logs

### Steps To Reproduce

1. Run from source (Main/head as of today) OR visit staging env
2. Go to the overview of any device e.g. `https://<FF_NAME_OR_IP>/device/<DEVICE_ID>/overview`

### Environment

- FlowForge version: Current - live on staging as of writing (not yet live on Production)
- Platform/OS: Windows
- Browser: Chrome

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

**Comment by Steve-Mcl** [comment]
UPDATE:

I have found the issue. 

Navigation entries need a `to` property to be set.  

Currently, the navigation entry for "Device logs" has a `path` property.  Changing [this](https://github.com/flowforge/flowforge/blob/e225b7457d6683bb148f5b5969ac404e3e78b57f/frontend/src/pages/device/index.vue#L150) to `to` fixes the issue.

**Comment by joepavitt** [comment]
It's now "to" rather than "path" as a property. It _should_ support async setting of tabs, suspect the main issue is the change in key/property.

**Comment by Steve-Mcl** [comment]
Yeah, I can confirm the component does work (even dynamically) when the `to` prop is provided. 

Unless we are confident there are no more of these gremlins, it may be want to consider supporting both `path` and `to` in the component?

**Comment by Steve-Mcl** [comment]
Also, I noted the component has a `featureUnavailable` property. Perhaps instead of adding the navigation entry dynamically in JS it should be added in the HTML and utilise that?

**Comment by joepavitt** [comment]
> we might be want to consider supporting both path and to in the component

No harm in it

**Comment by joepavitt** [comment]
> Also, I noted the component has a featureUnavailable property.

Need to remind myself what this does on Tuesday. IIRC it's for advertising premium features, but want to make sure before I give a thumbs up

**Comment by Steve-Mcl** [comment]
@joepavitt 

This had to be addressed quickly today during weekly release due to unforeseen circumstances.

The simple fix of changing `from` to `to` as first discussed in [this comment](https://github.com/flowforge/flowforge/issues/2620#issuecomment-1685338429) was applied to get us over the hump.

A quick search of the code for more instances of dynamically adding nav items with the old `path` property threw up some peculiar findings (i.e. there are lots of `navigation.push({..., path: ""}` but these seemed to be working).   So there is a subtly somewhere that should ideally be fully understood and handled if necessary.

We may wish to evaluate the various dynamic tab/menu/list item "things" and settle on a convention to avoid future issues?

**Comment by knolleary** [comment]
Given the immediate issue is resolved, I'm closing this issue.

If there's a desire for a follow-up task, please raise a separate issue.
